packages/client/hmi-client/src/page/data-explorer/components/tera-preview-panel.vue

```diff
@@ -80,6 +83,10 @@ const props = defineProps({
 	source: {
 		type: String,
 		default: 'xDD'
+	},
+	datasetSource: {
```

Comment:
Why not leveraging `resourceType`?

Reply:
I tiptoed along the line between ripping apart code and just adding what I needed to work, and this is one of thsoe cases. `ResourceType` gets weird because it already has `dataset` in it, which I need in other places and adding a `EsgfDataset` would have resulted in a lot more code changes.
A similar issue can be seen with documents - in `ResourceType` there is an "xdd" entry but no "document" entry so i think that terarium documents have this as undefined?

The front end seems to have a lot of collections of `Dataset | Document | DocumentAsset | Model` but then struggles to figure out what the object types are so there is a lot of these bandaid solutions around.  I think that the long term solution would be on the back end in `TerariumAsset` adding an `assetType` field so that the front end can see what an object is... but that would have required me to regenerate all our models, doing migrations and rewriting a lot of front end code... that feels like a "todo task" :D